src/libsphinxbase/feat/cmn_adapt.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * 16-Feb-2016  Zamir Ostroukhov <zamiron@gmail.com>
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#ifdef _MSC_VER
+#pragma warning (disable: 4244)
+#endif
+
+#include "sphinxbase/ckd_alloc.h"
+#include "sphinxbase/cmn.h"
+
+void
+cmn_adapt(cmn_t *cmn, mfcc_t **incep, int32 varnorm, int32 nfr) {
+
+    int32 i, j;
+    mfcc_t cep_cur[cmn->veclen];        // check it please, is it correctly?
+
+    if (nfr <= 0)
+        return;
+
+    for (j = 0; j < cmn->veclen; j++) {
+        cep_cur[j] = 0.0f;
+        for (i = 0; i < nfr; i++)
+            if ( abs(incep[i][j]) > cep_cur[j] )
+                cep_cur[j] = abs(incep[i][j]);
+
+        if ( cep_cur[j] == 0.0f )
+            cep_cur[j] = 0.0001f;
+
+        if ( cmn->max[j] == 0.0f )
+            cmn->max[j] = cep_cur[j] * 1.1f;
+
+        if ( cep_cur[j] > cmn->max[j] * 2.0f ) {
```

Comment:
2.0f is also not very reasonable constant here. Usually in proper code all constants are named to explain their meaning, for example:

`#define MAX_SCALE 2.0f`